codegen-compiler/src/main/kotlin/com/freeletics/khonshu/codegen/codegen/nav/NavDestinationModuleGenerator.kt

```diff
@@ -60,4 +70,25 @@ internal class NavDestinationModuleGenerator(
             }
         }
     }
+
+    private fun providesComponentProvider(): FunSpec {
+        return FunSpec.builder("bindComponentProvider")
+            .addAnnotation(provides)
+            .addAnnotation(intoMap)
+            .addAnnotation(mapKeyAnnotation())
+            .returns(
+                componentProvider.parameterizedBy(
+                    STAR,
+                    STAR,
+                ).copy(annotations = listOf(AnnotationSpec.builder(JvmSuppressWildcards::class).build())),
+            )
+            .addStatement("return %T", componentProviderClassName)
+            .build()
+    }
```

Comment:
@gabrielittner Do you think it would make sense to add generated code example in comments above the function? IMO that would make it easier to read.

Reply:
I can make sense but it wouldn't be maintainable. Even the tests are annoying to update for some tests